app/components/base_component.rb

```diff
@@ -1,7 +1,17 @@
 # frozen_string_literal: true
 
 class BaseComponent < ViewComponent::Base
+  def self.translation_key
+    @translation_key ||= name.snakecase.gsub("/", ".")
+  end
+
   def component_form_with(**options, &block)
     form_with(**options.merge(builder: GOVUKDesignSystemFormBuilder::FormBuilder), &block)
   end
+
+  def t(key, **options)
+    return super unless key.start_with?(".")
+
+    super(self.class.translation_key + key, **options)
+  end
```

Comment:
@kicferk1 This code just mimics how I1n works in views - it will only use the scope if the key starts with a `.`.

So `t('.label') will use a default search scope, however you can still access any translation by omiting the dot: `t('some_other_scope.inner.label`)`

ViewComponents have their own translation module, but I am a bit unsettled on how it works, as it expects a separate translation file in the same folder as the component (which I like), but it also isolates component translations, so it is no longer possible to access external translation (which I seriously do not like).

It is also worth noting that `t` is available within the component even without the code above. There was two problems with it though. Firstly, in their infinite wisdom, rails team decided to default translation scope to a physical, relative path of the erb file and I prefer we can move the files around without much fuff. Second problem was accessing those translations in the easy way in the test. Again, rails translation default scope is set on render, and since most of the tests are set up before render occurs, it was impossible to access translations without hardcoding the default scope.

Reply:
Ehh, alright. Yet another hidden cost of i18n that I actually never thought of, "where do we put the strings" :D